### ISSUE TITLE
Cablecam App issue

### ISSUE THREAD

**Issue by DmitriySurin** [Willyfan/cablecam-Setup#1]
First of all what I supposed to put  at the meters \ pulse dialog window at startup?

**Comment by DmitriySurin** [comment]
![screenshot_20180913-010514](https://user-images.githubusercontent.com/41476557/45455931-80424380-b6f1-11e8-908b-cbcdaa7d8d98.jpg)

**Comment by Willyfan** [comment]
at startup (only the first time) you must put the number of pulse you have from your cablecam in one meter. It depends from how much magnets you have on your wheel, the diameter and so on.
You can check it reading the position, move the camera by hand one meters and read the difference.
About the app, In old version I was a reading every one second, in last version I accelerate the reading every 250 ms. Maybe this is a problem with some device, so I rebult a version with one second.

**Comment by Willyfan** [comment]
beside, I not understand the error that you receive. Is when you connect the app to the cablecam?

**Comment by DmitriySurin** [comment]
Yes it is

**Comment by Willyfan** [comment]
Ok. Try the version I uploaded, same link and let me know.

**Comment by DmitriySurin** [comment]
Same error but some information I can recieve

**Comment by DmitriySurin** [comment]
![screenshot_20180913-011645](https://user-images.githubusercontent.com/41476557/45456340-0f9c2680-b6f3-11e8-9ff4-c568b8556ee7.jpg)

**Comment by DmitriySurin** [comment]
Please put the link to the app here

**Comment by Willyfan** [comment]
https://william-fanelli.com/wp-content/uploads/myupload/ID_CablecamSetup_8.apk

**Comment by Willyfan** [comment]
the app receive all the info in the correct format. Maybe there is some geo difference in use data delimiter. Where are you?

**Comment by DmitriySurin** [comment]
Russia

**Comment by Willyfan** [comment]
Ok, in  Russia separator for decimal number is point or comma? on your phone?

**Comment by DmitriySurin** [comment]
I gues point

**Comment by DmitriySurin** [comment]
But not shure what you mean

**Comment by DmitriySurin** [comment]
In calculator app i can see comma

**Comment by Willyfan** [comment]
I'm building a versione than don't calculate position and limits in metres, may it work. please dowload and try

**Comment by Willyfan** [comment]
A question, but do you put something in the pulse/metres value? because if you don't put a number maybe it don't work

**Comment by DmitriySurin** [comment]
Ok the problem was if I skip strtup dialog and put nothing that cause the problem. I put 1 for example and everything start to work as expected...

**Comment by DmitriySurin** [comment]
Got it)

**Comment by DmitriySurin** [comment]
What about rope force calculation?

**Comment by Willyfan** [comment]
Ok. Now I put a default value if someone don't put a number. thank you

**Comment by Willyfan** [comment]
Rope force calculator is ok.

**Comment by DmitriySurin** [comment]
Forces
The formula to calculate the rope force is shown here


Let's try a simple calculation.

The distance to cover is l=100m. The weight of our cablecam is p=3kg. The tolerated maximum slack when in the middle is f=1m.

sqrt( (3/2)² + (3*100/ (4*1)² ) = sqrt ( 1.5² + 75² ) = 75kg

**Comment by DmitriySurin** [comment]
Your app shows 125 kg

**Comment by DmitriySurin** [comment]
![screenshot_20180913-014208](https://user-images.githubusercontent.com/41476557/45457162-67885c80-b6f6-11e8-922c-bc1e0adab5d0.jpg)

**Comment by DmitriySurin** [comment]
Rope parameters doesnt affect the rope force calculation

**Comment by Willyfan** [comment]
ok, I will check tomorrow

**Comment by DmitriySurin** [comment]
Also autoplay works only 4-5 cucles...after that the app is freezing(

**Comment by DmitriySurin** [comment]
![screenshot_20180913-020216](https://user-images.githubusercontent.com/41476557/45457811-61e04600-b6f9-11e8-852b-508324c6fe74.jpg)

**Comment by DmitriySurin** [comment]
What value set to 1?

**Comment by Willyfan** [comment]
I found the bug, I'm using only one type of rope. Now is working.
About play button, there is some problem with bluetooth, I working on a windows version.
If you like I try to fix also in this app, but it is unusable in real world.

**Comment by Willyfan** [comment]
But this tomorrow, now is night...

**Comment by DmitriySurin** [comment]
SO good night)

**Comment by Willyfan** [comment]
Download and try last build.
Now, I changed time for refresh bluetooth receiver at 500 ms.
At startup, if you don't input a correct value you receive a warning, and the pulse/meter is set to a default value.
Play function don't display anymore the alert "value set to 1", so it don't freeze. You can see at bottom the log for all commands anyway. The commmad $p 1 is sent every 1600ms. When you stop play, the command is not more sent but the cablecam wait 2000 ms from last command before stop because I don't send the stop command ($P 0) for now.

**Comment by DmitriySurin** [comment]
A checked autoplay...the cablecam stop itself and the app is freeze after about 15-20 cycles. Also have you try to engage autoplay via radio?

**Comment by Willyfan** [comment]
Yes, I use it always by radio. The problem in bluetooth is that it don't have a long range, and the cablecam stop it if don't receive command from bluetooth.
Beside, I checked autoplay for around one minutes without freeze, but I'm sure that I was in bluetooth range (laboratory).
If you want don't use a radio you must go on other communication system, like telemetry like this one:
https://www.ebay.it/itm/1-Set-3DR-Radio-Telemetry-Kit-433Mhz-Module-Open-Source-per-APM-2-5-2-6-2-8/253259457452?hash=item3af770a3ac:g:vnEAAOSwp-Rady2Z
(there are version at 433 mhz and 900 mhz, it depend in your country what is permitted, in Italy only 433).
Actually, i'm unable to write a app on android for use this radios (but I know that is possible), But I'm writing a windows app that can use it:
![winapp](https://user-images.githubusercontent.com/36896839/45490942-b4a51680-b768-11e8-9d3e-69eca0b8021a.jpg)

With this system you can have total control of cablecam from a ground station

**Comment by DmitriySurin** [comment]
Cool man) But my phone was always near the cablecam. I spin encoder by hand... so there is no problem in range but in cade may be...btw what kind of motor you use?

**Comment by DmitriySurin** [comment]
> I found the bug, I'm using only one type of rope. Now is working.
But actually also I dont understand why the formula from Daen page show different result that you app even if I use 4mm w/cover. And btw would be very handy if you add dyneema type of rope to you calculator.

**Comment by Willyfan** [comment]
Formula by Werner don't take account of cable weight. In 100 metres you must add half of total cable weight, so is 2 kg for a 4 mm dyneema rope. If you have a cablecam that weight 3 kg, adding 2 kg in a lot more and results change a lot.
Beside, all kind of cable in my app are dyneema. Dyneema is a commercial name, and HMPE (high-modulus polyethylene) is a more generic name. Maybe there is little difference from HMPE product from some producer, but is around 2 or 3%, is not important for our calculation.
Difference from cables are only:
1) diameters
2) with or without cover

**Comment by DmitriySurin** [comment]
Ok understand. What does mean cover or wo cover?

**Comment by Willyfan** [comment]
Normally the rope is a core in dyneema with a protection external cover in polyester.
As you can imagine, if total diameter is 8 mm dyneema is, for example, 6 mm. 
A rope without cover is all dyneema, without external cover. You can have same resistence with lower diameter and weight, or more resistence at same diameter. But cost is higher, and the core is unprotected and it have less grip.

**Comment by DmitriySurin** [comment]
Btw what kind of motor you use?

**Comment by Willyfan** [comment]
I'm using a alien power 5065 270kv (or 220Kv)

**Comment by Willyfan** [comment]
> Btw what kind of motor you use?

Beside, download the last version of app, last night I don't see a error on rope weight calculation. 40 kg for 100 mt of 8 mm dyneema is a lot...

**Comment by Willyfan** [comment]
new app and new firmware are online
